tests/utils/test_gen_detector_sal.py

```diff
@@ -0,0 +1,191 @@
+import numpy as np
+from typing import Iterable, Tuple, Dict, Hashable
+
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+from xaitk_saliency.interfaces.gen_detector_prop_sal import GenerateDetectorProposalSaliency
+from xaitk_saliency.utils.tools.gen_detector_sal import gen_detector_sal
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_detection.utils.bbox import AxisAlignedBoundingBox
+
+
+class TestGenDetectorSal:
+
+    def test_sal_generation(self) -> None:
+        """
+        Test saliency generation functionality with dummy images, detector, perturber, and saliency generator.
+        """
+        class TestDetector(DetectImageObjects):
+            """
+            Dummy detector that returns constant detections
+            """
+            def detect_objects(
+                self,
+                img_iter: Iterable[np.ndarray]
+            ) -> Iterable[Iterable[Tuple[AxisAlignedBoundingBox, Dict[Hashable, float]]]]:
+                for i, img in enumerate(img_iter):
+                    yield [(AxisAlignedBoundingBox((0, 0), (100, 100)), {0: 1.0})] * i
+
+            def get_config(self) -> dict:
+                return {}
```

Comment:
Since this and other similar methods defined here are not used, and we don't expect them to be, we can define here as
```
def get_config(self) -> dict: ...
```

I might also suggest defining them outside of the test method for code clarify reasons.

This change would address the codecov "failure" where the tests are not themselves being "covered" 100%.